Fixed ansible run issue when hostname has trailing dot

jonnyshaw89 opened the pull request:
### What is the context of this PR?
The Ansible scripts were failing silently when the hostnames in the `ansible-playbook` command had a trailing dot.
This was very susceptible to user error.
The new solution is not susceptible and uses the FQDN from the DNS record.

Also Fixes https://github.com/ONSdigital/eq-terraform/issues/88

### How to review
Deploy the infrastructure on AWS.
Run through a survey and submit.
The survey should be submitted successfully.

jonnyshaw89 committed:
Fixed ansible run issue when hostname has trailing dot

jonnyshaw89 committed:
Enable Detailed Monitoring

jonnyshaw89 committed:
Remove duplicated alarm and use count

jonnyshaw89 committed:
Treat missing data as breaching

jonnyshaw89 committed:
Tidy up Alerting

adoublebarrel commented:
I'm unable to provision this without getting the man in the middle attack warning when ansible tries to install rabbit. That's with the ANSIBLE_HOST_KEY_CHECKING env var set to false as well. Have I missed something?

jonnyshaw89 committed:
Increase cluster status alarm to 2 evaluation_periods

Commit:
Removed control_path from Ansible config

It was causing a '[string] too long for unix domain socket'
error. After some googling it turns out control path was often used as
a workaround for this error but now a proper fix has gone in and
having it can actually cause the error. The recommendation is to
remove it which worked for me. See the end of https://github.com/ansible/ansible/issues/11536.

jonnyshaw89 committed:
Opened Port for SDX

Commit:
Merge branch 'fix-ansible-rabbitmq-provisioning' of github.com:ONSdigital/eq-terraform into fix-ansible-rabbitmq-provisioning

Commit:
Disabled Ansible host key checking

Added the environment variable to the front of the playbook command.

adoublebarrel commented:
So ANSIBLE_HOST_KEY_CHECKING only disables the prompt for adding servers to your .ssh/known_hosts file. If the id for those hosts changes then you'll get the man in the middle attack warning. Solution is to delete the rabbit queues from the known_hosts file.

jonnyshaw89 committed:
Remove ansible.cfg as it did nothing

jonnyshaw89 commented:
I Approve the changes made by @adoublebarrel

Commit:
Merge branch 'master' into fix-ansible-rabbitmq-provisioning
